src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1777,6 +1779,44 @@ internal void EnsureNativeIntegerAttributeExists()
             EnsureEmbeddableAttributeExists(EmbeddableAttributes.NativeIntegerAttribute);
         }
 
+#nullable enable
+        /// <summary>
+        /// Creates the ThrowIfNull and Throw helpers if needed.
+        /// </summary>
+        /// <remarks>
+        /// The ThrowIfNull and Throw helpers are modeled off of the helpers on ArgumentNullException.
+        /// https://github.com/dotnet/runtime/blob/22663769611ba89cd92d14cfcb76e287f8af2335/src/libraries/System.Private.CoreLib/src/System/ArgumentNullException.cs#L56-L69
+        /// </remarks>
+        internal MethodSymbol EnsureThrowIfNullFunctionExists(SyntaxNode syntaxNode, SyntheticBoundNodeFactory factory, DiagnosticBag? diagnostics)
+        {
+            var privateImplClass = GetPrivateImplClass(syntaxNode, diagnostics);
+            lock (privateImplClass)
```

Comment:
>lock (privateImplClass) [](http://example.com/codeflow?start=12&length=23)

Could we avoid the lock by always using the symbols from `PrivateImplementationDetails`?
```csharp
if (privateImplClass.GetMethod("ThrowIfNull") is null)
{
    privateImplClass.TryAddSynthesizedMethod(new SynthesizedThrowMethod(...));
    privateImplClass.TryAddSynthesizedMethod(
        new SynthesizedThrowIfNullMethod(...
            privateImplClass.GetMethod("Throw"));
}
return privateImplClass.GetMethod("ThrowIfNull");
```

Reply:
I guess this does guarantee we would use the ThrowMethod that really got added :)